### PR TITLE
Use YAML based columns config

### LINKED ISSUE #3: Improve columns mapping structure
In BOM generation, we've found that the columns mapping is hard for people to understand at first. The structure of the columns mapping - a dictionary with the keys being the name of the columns and the value being a list of attributes to check - has a lot of implicit meaning that makes it easy for us as developers to work with but harder to understand. 

Since we are changing the language for the configuration, we could also change the columns mapping to look more like:

```yaml

columns:
  - name: Designator
    attributes_to_check: Designator
  - name: Manufacturer
    attributes_to_check:
      - Manufacturer
      - MFN
  - name: Part Number
    attributes_to_check:
      - Part
      - PN
      - _part_id
```

### PR DESCRIPTION
**Pull request by shrik450**
Closes #3. 

See https://test.allspice.dev/app_path/AllSpice/actions-test/pulls/15 for a run with the new type of columns config.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,89 +16,109 @@ Add the following steps to your actions:
...
+about YAML, [check out the AllSpice Knowledge Base.](https://learn.allspice.io/docs/yaml)
+
+The format of this YAML file is as follows:
+
+```yml
+columns:
+  - column_name: Manufacturer
+    part_attributes:
+      - Manufacturer
+      - MANUFACTURER
```
Comment (jtran) [inline README.md:50]: `column_name` is already in the context of `columns`. Should it be just `name`?

File: README.md [hunk 2]
```diff
@@ -16,89 +16,109 @@ Add the following steps to your actions:
...
+  - column_name: Part Number
+    part_attributes:
+      - PART
+      - MANUFACTURER \#
```
Comment (kdumontnu) [inline README.md:54]: Unfortunately, I think escaping special characters is going to be a footgun for any users learning YAML. Should we suggest quoting all strings? 

We probably want to link to some document page on getting started w/ yaml, like https://docs.ansible.com/ansible/latest/reference_appendices/YAMLSyntax.html

**Review by kdumontnu — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (shrik450) [reply README.md:54]: I've linked the AllSpice Knowledge Base article (https://learn.allspice.io/docs/yaml) above, but I can change it to this link if you think that's better.

File: README.md (hunk 1 above)
Comment (shrik450) [reply README.md]: Changed.

**Review by jtran — APPROVED** [review]

**Comment by shrik450** [comment]
I'll merge this, and if we want to make any more doc changes we can track that separately.

**Pull request merged**